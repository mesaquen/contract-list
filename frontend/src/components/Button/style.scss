.button {
  padding: 0.5rem 1rem;
  border-width: 2px;
  border-color: transparent;
  border-style: solid;
  font-size: 0.875rem;
  border-radius: var(--roundness);
  background-color: var(--white);
  cursor: pointer;
  font-weight: 600;
  color: var(--primary-main-titles);

  &:hover {
    background-color: var(--white-hover);
    box-shadow: var(--shadow);
  }

  &:focus {
    outline: none;
  }

  transition: all 0.3s ease-in-out;
}

.raised {
  border-color: var(--primary-yellow);
  box-shadow: var(--shadow);
}

.primary {
  background-color: var(--primary-yellow);

  &:hover {
    background-color: var(--primary-hover);
    border-color: var(--primary-hover);
  }
}

.disabled {
  background: var(--dividers-bg-bg-menu);
  color: var(--tertiary-field-borders);
  border-color: var(--disabled-border);
  box-shadow: none;
  &:hover {
    background-color: var(--dividers-bg-bg-menu);
    cursor: unset;
    box-shadow: none;
  }
}
