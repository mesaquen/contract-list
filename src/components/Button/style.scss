@import '../../shared/commonStyles.scss';

.button {
  padding: 0.5rem 1rem;
  border-width: 2px;
  border-color: transparent;
  border-style: solid;
  font-size: 0.875rem;
  border-radius: var(--roundness);
  background-color: var(--white);
  cursor: pointer;
  font-weight: 600;
  color: var(--primary-main-titles);

  &:hover {
    background-color: var(--white-hover);
    box-shadow: var(--shadow);
  }

  &:focus {
    outline: none;
  }

  transition: all 0.3s ease-in-out;
}

.raised {
  border-color: var(--primary-yellow);
  box-shadow: var(--shadow);
}

.primary {
  background-color: var(--primary-yellow);

  &:hover {
    background-color: var(--primary-hover);
    border-color: var(--primary-hover);
  }
}
