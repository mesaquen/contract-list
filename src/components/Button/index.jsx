import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'
import './style.scss'

const Button = ({ children, raised, primary, ...props }) => {
  const classNames = cx(['button'], {
    raised: raised || primary,
    primary,
  })

  return (
    <button className={classNames} {...props}>
      {children}
    </button>
  )
}

Button.propTypes = {
  primary: PropTypes.bool,
  raised: PropTypes.bool,
  children: PropTypes.node,
}

export default Button
