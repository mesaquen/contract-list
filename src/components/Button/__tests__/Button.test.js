import Button from '../index'
import { cleanup, fireEvent, render, screen } from '@testing-library/react'

beforeEach(cleanup)
describe('Button', () => {
  test('callback is called when clicked', async () => {
    const callback = jest.fn(() => null)
    render(
      <Button data-testid='button' onClick={callback}>
        sample
      </Button>
    )
    const element = await screen.findByTestId('button')
    fireEvent.click(element)
    expect(callback.mock.calls.length).toBe(1)
  })

  test('text is rendered', async () => {
    render(
      <Button data-testid='button'>
        sample
      </Button>
    )
    const element = await screen.findByText('sample')
    expect(element).toBeInTheDocument()
  })

  test('match snapshot when raised',  () => {
      const {asFragment} = render(<Button raised>RaisedButton</Button>)
      expect(asFragment()).toMatchSnapshot()
  })

  test('match snapshot when primary',  () => {
    const {asFragment} = render(<Button primary>PrimaryButton</Button>)
    expect(asFragment()).toMatchSnapshot()
})
})
