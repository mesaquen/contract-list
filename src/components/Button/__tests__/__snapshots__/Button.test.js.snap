// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button match snapshot when primary 1`] = `
<DocumentFragment>
  <button
    class="button raised primary"
  >
    PrimaryButton
  </button>
</DocumentFragment>
`;

exports[`Button match snapshot when raised 1`] = `
<DocumentFragment>
  <button
    class="button raised"
  >
    RaisedButton
  </button>
</DocumentFragment>
`;
